### PR TITLE
Statement exhausted compilation resources [sc-13029]

### PR DESCRIPTION
**Pull request by alyiwang**
### 🤔 Why?

For a snowflake database with a large number of tables, the query to get table info including DDL can get exceedingly long. 

### 🤓 What?

- Cap the number of tables info to fetch in one query to 100

### 🧪 Tested?

tested against snowflake instance, and verified the output is the same as before.

### CODE REVIEW

**Review by mars-lan — COMMENTED** [review]
Bump version?

File: metaphor/snowflake/extractor.py [hunk 1]
```diff
@@ -213,10 +217,19 @@ def _fetch_columns(self, cursor: SnowflakeCursor, database: str) -> None:
...
 
-    def _fetch_table_info(self, tables: Dict[str, DatasetInfo], shared: bool) -> None:
+    def _fetch_table_information(
+        self, tables: Dict[str, DatasetInfo], shared: bool
+    ) -> None:
+        chunks = utils.chunks([item for item in tables.items()], TABLE_INFO_FETCH_SIZE)
+        for chunk in chunks:
+            self._fetch_table_info(chunk, shared)
+
+    def _fetch_table_info(
```
Comment (mars-lan) [inline metaphor/snowflake/extractor.py:227]: Consider rename this to `_fetch_table_info_internal` or make it an inner function?

Comment (alyiwang) [reply metaphor/snowflake/extractor.py]: updated

Comment (mars-lan) [reply metaphor/snowflake/extractor.py]: My bad. Given how long the function is, it's not so readable after making it an inner function. Maybe just rename it?

**Review by mars-lan — DISMISSED** [review]

File: metaphor/snowflake/extractor.py (hunk 1 above)
Comment (alyiwang) [reply metaphor/snowflake/extractor.py]: reverted.

**Review by mars-lan — APPROVED** [review]

**Pull request merged**